ui/app/milestone/factories/get-milestones.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+(function () {
+
+    var module = angular.module('pnc.milestone');
+
+    module.factory('versionFactory', ['$http',
+
+        function ($http) {
+            var restUrl = '../pnc-rest/rest/product-milestones/product-versions/',
+                factory = {};
+
+            factory.checkUniqueValue = function (productVersionId, milestoneVersion, productVersion) {
+                return $http.get(restUrl + productVersionId, {cache: true}).then(
```

Comment:
It's better that you inject our REST client object here, our base URLs are dynamically configured using the PNC main config file and this will take care of it for you. You can do this by:
- Inject ProductMilestoneDAO
- Call ProductMilestoneDAO.getByProductVersion({ versionId: productVersionId }).$promise.then( ....